### PR TITLE
Tools: Topology: Updated TDFB macro files

### PR DESCRIPTION
**Pull request by singalsu**
The tdfb_controls.m4 is changed for better 15 degree control enum
spacing for beam direction. The file is with this change dedicated
for line arrays with physical limitation to -90 to +90 degrees that
is just achievable with 16 max. enum values.

The new tdfb_controls360.m4 uses the previous 30 degree spacing for
360 degrees arrays such as circular. The pipeline macros for 360
degrees are added to pipelines macros. The difference is only include
of 360 degrees controls version.

The topology development CMakeList.txt is changed to build topologies
with TDFB with single dual mono -90 to +90 degree angle steerable
beam. The stereo dual beam setup files are preserved and updated but
currently used only for testbench test for more complex configuration.

Signed-off-by: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>

### CODE REVIEW

**Review by lgirdwood — COMMENTED** [review]
@singalsu I'm assuming all the topologies have been updated using a tool ?

**Comment by singalsu** [comment]
> @singalsu I'm assuming all the topologies have been updated using a tool ?

Yep that is correct. The commands sequence in tools/tune/tdfb from Matlab to produce these is:

```
example_pass_config
example_line_array
example_line_0mm36mm146mm182mm
example_circular_array
example_two_beams
```

**Comment by lgirdwood** [comment]
> > @singalsu I'm assuming all the topologies have been updated using a tool ?
> 
> Yep that is correct. The commands sequence in tools/tune/tdfb from Matlab to produce these is:
> 
> ```
> example_pass_config
> example_line_array
> example_line_0mm36mm146mm182mm
> example_circular_array
> example_two_beams
> ```

Thanks, can we mention this in the TDFB file (in the comments) so that anyone can update the coefficients. Thanks.

**Comment by singalsu** [comment]
> > > @singalsu I'm assuming all the topologies have been updated using a tool ?
> > 
> > 
> > Yep that is correct. The commands sequence in tools/tune/tdfb from Matlab to produce these is:
> > ```
> > example_pass_config
> > example_line_array
> > example_line_0mm36mm146mm182mm
> > example_circular_array
> > example_two_beams
> > ```
> 
> Thanks, can we mention this in the TDFB file (in the comments) so that anyone can update the coefficients. Thanks.

Yep, I think I'll add a script example_all.sh that generates those. There's some waiting in each step and it's a rather boring task to do without script. And refer to script in tdfb.c!

**Review by lgirdwood — APPROVED** [review]

**Comment by singalsu** [comment]
@lgirdwood I added the example_all.sh script into this patch. In other C code PR I added a commit that adds the note about settings generator script.

**Review by marc-hb — COMMENTED** [review]
You could remove all the duplication like this:

```shell
  
  # arrays are whitespace and special character safe and shellcheck compliant
  # https://mywiki.wooledge.org/BashGuide/Arrays
  OCTAVE_CMD=( octave --no-gui --quiet )
  ...
  elif command -v octave &> /dev/null; then
       CMD=( "${OCTAVE_CMD[@]" )
       EXT='.m'
   else
   ...
   fi
   run_examples
}

# Now this is easy to change, temporarily or permanently
TESTS_LIST=( _pass_config  _line_array ... )

run_examples()
{
  for t "${TESTS_LIST[@]}" ; do
   "${CMD[@]}"  example_"${t}""$EXT"
  done
}

File: tools/tune/tdfb/example_all.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+set -e
+
+OCTAVE_CMD="octave --no-gui --quiet"
+MATLAB_CMD="matlab -batch"
+
+main () {
+    if command -v matlab &> /dev/null; then
+	echo "Using Matlab"
+	use_matlab
+	exit 0
```
[the remark this answers is not in the document]
Comment (singalsu) [reply tools/tune/tdfb/example_all.sh]: Done, thanks for nice batch example!

**Comment by singalsu** [comment]
> `example_all.sh` looks good to me now but that's a tiny part of the whole PR.

The non-machine generated topology changes are in the last fifth of the scroll window (none of the hidden diffs opened).

**Comment by lgirdwood** [comment]
@XiaoyunWu6666 @keqiaozhang CI link is not showing any results here. Can you check thanks. @mengdonglin fyi.

**Comment by fredoh9** [comment]
> @XiaoyunWu6666 @keqiaozhang CI link is not showing any results here. Can you check thanks. @mengdonglin fyi.
> 
> https://sof-ci.01.org/sofpr/PR5288/build11932/devicetest

I checked the logs and test results output. But there is no error in our side. For the test results page, we depend on external library/script/css like vue, I guess it is broken from there.

**Comment by lgirdwood** [comment]
> > @XiaoyunWu6666 @keqiaozhang CI link is not showing any results here. Can you check thanks. @mengdonglin fyi.
> > https://sof-ci.01.org/sofpr/PR5288/build11932/devicetest
> 
> I checked the logs and test results output. But there is no error in our side. For the test results page, we depend on external library/script/css like vue, I guess it is broken from there. I will search for similar problem.

Working now.

**Pull request merged**